app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsLocation.kt

```diff
@@ -1,21 +1,71 @@
 package ch.epfl.sdp.blindly.settings
 
+import android.location.Location
 import android.os.Bundle
 import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 import ch.epfl.sdp.blindly.R
+import ch.epfl.sdp.blindly.location.AndroidLocationService
+import ch.epfl.sdp.blindly.location.LocationService
+import com.google.android.gms.maps.*
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
 
-class SettingsLocation : AppCompatActivity() {
+
+class SettingsLocation : AppCompatActivity(), OnMapReadyCallback {
+
+    private lateinit var mapView: MapView
+    private var location: Location? = null
+    private lateinit var locSer: LocationService
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_settings_location)
 
-        supportActionBar?.hide()
+        locSer = AndroidLocationService(this)
+        location = locSer.getCurrentLocation()
 
         val currentLocation = intent.getStringExtra(EXTRA_LOCATION)
         findViewById<TextView>(R.id.my_current).apply {
             text = currentLocation
         }
+
+        supportActionBar?.hide()
+
+        mapView = findViewById(R.id.map)
+        mapView.onCreate(savedInstanceState)
+        mapView.getMapAsync(this)
+
+    }
+
+    override fun onMapReady(googleMap: GoogleMap) {
+        val loc: LatLng = if (location != null) {
+            LatLng(location!!.latitude, location!!.longitude)
+        } else {
+            LatLng(46.5, 6.6)

```

Comment:
What are these latitude and longitude values, can they be constants ?

Reply:
I agree they could have been contants (it's the latitude and longitud of lausanne)